content/post/2021-12-01-mlr3spatial/index.Rmd

```diff
@@ -0,0 +1,30 @@
+---
+title: mlr3spatial
+author: Marc Becker
+date: '2021-12-01'
+slug: mlr3spatial
+categories:
+  - R
+tags:
+  - mlr3spatial
+  - CRAN
+---
+
+We are happy to announce that [mlr3spatial](https://mlr3spatial.mlr-org.com/) has been released on [CRAN](https://cran.r-project.org/web/packages/mlr3spatial/index.html). 
+[mlr3spatial](https://mlr3spatial.mlr-org.com/) simplifies the handling of spatial objects in the [mlr3](https://github.com/mlr-org) ecosystem.
+Before [mlr3spatial](https://mlr3spatial.mlr-org.com/), the user had to extract raw values from spatial objects to train a model or predict a map.
+
+Now, mlr3 [tasks](https://mlr3.mlr-org.com/reference/Task.html) can directly read from spatial objects via specialized [data backends](https://mlr3.mlr-org.com/reference/DataBackend.html).
+Such tasks can be used to train a model or to perform resampling just like any other mlr3 task.
+We support spatial raster objects created by the packages [terra](https://CRAN.R-project.org/package=terra), [raster](https://CRAN.R-project.org/package=raster) and [stars](https://CRAN.R-project.org/package=stars).
+Additionally, vector data is handled with the package [sf](https://CRAN.R-project.org/package=sf).
+
+The `predict_raster()` function makes a map with predictions from an mlr3 learner.
+We only have to pass a task with a spatial data backend which provides the data and spatial reference.
+To avoid memory issues with large raster files, the prediction is done in chunks.
+One after the other, the raster chunks are loaded into memory and the prediction is written to disk.
+Finally, the complete map is available on disk. 
+The learner can also make use of future-based parallelization to accelerate the predictions.
```

Comment:
We might want to explain in more detail how the chunking works and how the user can control it (-> argument) and what the default is doing.

Reply:
The user can adapt the chunks size (`chunksize`) to the available memory.
The default chunk size is 200 Megabytes which should work on most consumer consumer grade machines.
If more memory is available, a larger chunk size accelerates the prediction process. 

?

Reply:
I think the term "chunk size" needs to be explained/introduced in more detail. 
Also we should mention that this behavior might change if {terra} changes something WRT to memory usage etc. 

We should also explain we have chosen 200 MB as most consumer machines have more available memory. 

> If more memory is available, a larger chunk size accelerates the prediction process.

Yes, we should add that this needs to be enforced by the user.